lms/templates/navigation.html

```diff
@@ -21,6 +20,26 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
+% if uses_pattern_library:
+    <%block name="js_extra">
+    <%static:require_module module_name="js/header_factory" class_name="HeaderFactory">
+    HeaderFactory();
+    </%static:require_module>
+    </%block>
+% endif
+
+<%
+if uses_pattern_library:
```

Comment:
I'm concerned that the HTML having special knowledge of which stylesheet is currently applied is going to be difficult to maintain, since styles and markup should be free to change over time. How about we agree on a common set of class names (they can be the ones implemented by the UXPL), and make sure that both stylesheets implement them? Then we won't need all this branching within the templates.